tls/s2n_connection.c

```diff
@@ -357,10 +357,23 @@ int s2n_connection_get_delay(struct s2n_connection *conn)
 int s2n_sleep_delay(struct s2n_connection *conn)
 {
     if (conn->blinding == S2N_BUILT_IN_BLINDING) {
-        int delay;
+        int delay, r;
         GUARD(delay = s2n_connection_get_delay(conn));
-        GUARD(sleep(delay / 1000000));
-        GUARD(usleep(delay % 1000000));
+
+        SLEEP:
+        r = sleep(delay / 1000000);
+        if (r != 0) {
+            goto SLEEP;
```

Comment:
according to the sleep manpage:

Return Value:

Zero if the requested time has elapsed, or the number of seconds left to sleep, if the call was interrupted by a signal handler.

so, in that case, should we set delay = (r + (delay % 1000000)) in the case of a non-zero return value?